imports/plugins/core/components/lib/blocks.js

```diff
@@ -0,0 +1,274 @@
+ * @param {String} name The name of the component to register.
+ * @param {React.Component} newComponent Interchangeable/extendable component.
+ * @param {Function|Array} hocs The HOCs to compose with the raw component.
+ * @returns {Function|React.Component} A component callable with Components[name]
+ * @memberof Components/Helpers
+ */
+export function replaceBlock({ regionName, blockName, newBlock, hocs = [] }) {
+  const previousBlock = BlocksTable[regionName][blockName];
+
+  if (!previousBlock) {
+    throw new Error(`Block '${name}' of region ${regionName} not found. Use registerComponent to create it.`);
+  }
+
+  const newHocs = Array.isArray(hocs) ? hocs : [hocs];
+
+  return registerBlock({
+    name,
+    component: newBlock,
+    hocs: [...newHocs, ...previousBlock.hocs]
+  });
+}
+
+
+/**
+ * @name getRawBlockComponent
+ * @method
+ * @summary Get the raw UI component without any possible HOCs wrapping it.
+ * @param {String} regionName The name of the block region.
+ * @param {String} blockName The name of the block component to get.
+ * @returns {Function|React.Component} A React component
+ * @memberof Components/Helpers
+ */
+export const getRawBlockComponent = (regionName, blockName) => BlocksTable[regionName][blockName].rawComponent;
+
+
+/**
+ * @name getBlockHOCs
+ * @method
+ * @summary Get the raw UI component without any possible HOCs wrapping it.
+ * @param {String} regionName The name of the block region.
+ * @param {String} blockName The name of the block component to get HOCs from.
+ * @returns {Function|React.Component} Array of HOCs
+ * @memberof Components/Helpers
+ */
+export const getBlockHOCs = (regionName, blockName) => BlocksTable[regionName][blockName].hocs;
+
+
+/**
+ * @name copyBlockHOCs
+ * @method
+ * @summary Wrap a new component with the HOCs from a different component
+ * @param {String} sourceRegionName The name of the block region
+ * @param {String} sourceBlockName The name of the block component to get the HOCs from
+ * @param {Function|React.Component} targetBlock Block component to wrap
+ * @returns {Function|React.Component} A new component wrapped with the HOCs of the source component
+ * @memberof Components/Helpers
+ */
+export function copyBlockHOCs(sourceRegionName, sourceBlockName, targetBlock) {
+  const sourceComponent = BlocksTable[sourceRegionName][sourceBlockName];
+  return compose(...sourceComponent.hocs)(targetBlock);
+}
+
+/**
+ * @name loadRegisteredBlocks
+ * @method
+ * @summary Populate the final BlockComponents object with the contents of the lookup table.
+ * This should only be called once on app startup.
+ * @returns {Object} An object containing all of the registered blocks by region
+ * @memberof Components/Helpers
+ **/
+export function loadRegisteredBlocks() {
+  Object.keys(BlocksTable).forEach((regionName) => {
+    BlockComponents[regionName] = getBlocks(regionName);
+  });
+  return BlockComponents;
```

Comment:
Is there a reason this needs to be run on startup? Couldn't everything in here and `getBlocks` just happen within `registerBlock` directly? I'd like to avoid adding startup code whenever possible.

Reply:
I believe I tried without this load components function but it resulted in the blocks being rebuilt, with their HOCs every time a change was saved to a field. That meant you couldn't alt-tab to the next field as the component would re-render and lose focus. This change + using JSX in the `Blocks` component seems to solve that issue.

Reply:
OK it's something we could look into and adjust later anyway.